tools/probes/probes_main.c

```diff
@@ -219,7 +219,7 @@ void parse_data(char *file_in)
 				/* request to copy full data packet */
 				total_data_to_copy = sizeof(struct probe_data_packet) /
 					sizeof(uint32_t);
-				w_ptr = (uint32_t *)packet;
+				w_ptr = __builtin_assume_aligned((uint32_t *)packet, 1);
```

Comment:
Btw, I assume we are aligning on 1 byte because we need to synchronise the input stream on the structure header (and this can be caused by interruptions in the input stream) ? If so, lets make this obvious in the commit message and as some inline comments (otherwise folks will see this a a performance bottleneck).

Reply:
The only issue is GCC warning because probe packet struct is forced to align 1 (by _packed_ attribute). Whatever  (1 or 4) we will put in __builtin_assume_aligned it will be ok for gcc

Reply:
Everything is 4 byte aligned looking at the code.

Reply:
@lgirdwood it is but the struct is forced to be "packed" so the alignment is 1.
Changed assumption to 4, both are okay for gcc